src/checksum.c

```diff
@@ -17,7 +17,7 @@ G_STATIC_ASSERT(RAUC_DEFAULT_CHECKSUM != 0);
 G_DEFINE_QUARK(r-checksum-error-quark, r_checksum_error)
 
 static gboolean
-update_from_file(GChecksum *ctx, const gchar *filename, gsize *total, GError **error)
+update_from_file(GChecksum *ctx, const gchar *filename, goffset *total, GError **error)
 {
 	g_auto(filedesc) fd = -1;
 	gsize size = 0;
```

Comment:
This needs to be changed to `goffset` as well as it's used to accumulate the number of processed bytes which is later assigned to `*total`

```suggestion
	goffset size = 0;
```

With that line changed I've been able to install a >2GB bundle on an armv7 system

Reply:
Indeed. Thanks!